controller/Controller.py

```diff
@@ -0,0 +1,207 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script implementing a remote controller for the DDS Recorder."""
+
+from enum import Enum
+
+from PyQt6.QtCore import QObject, pyqtSignal
+
+from ddstypes.ControllerCommand.ControllerCommand import (
+    ControllerCommand, ControllerCommandPubSubType)
+from ddstypes.Status.Status import Status, StatusPubSubType
+
+import fastdds
+
+from utils.Logger import Logger
+
+
+class DdsRecorderStatus(Enum):
+    RECORDING = 1
+    PAUSED = 2
+    STOPPED = 3
+    UNKNOWN = 4
+
+
+class CommandWriterListener(fastdds.DataWriterListener):
+    """Fast DDS DataWriter listener class for ControllerCommand DataWriter."""
+
+    def __init__(self, controller):
+        """Construct a WriterListener object."""
+        super().__init__()
+        self._controller = controller
+
+    def on_publication_matched(self, writer, info):
+        """Raise when there has been a match/unmatch event."""
+        if (0 < info.current_count_change):
+            self.logger.debug('DDS Recorder found!')
+            self._controller.on_ddsrecorder_discovery.emit(
+                'DDS Recorder found!')
+        else:
+            self.logger.debug('DDS Recorder lost!')
+            self._controller.on_ddsrecorder_discovery.emit(
+                'DDS Recorder lost!')
+
+
+class StatusReaderListener(fastdds.DataReaderListener):
+    """Fast DDS DataReader listener class for Status DataReader."""
+
+    def __init__(self, controller):
+        """Construct a WriterListener object."""
+        super().__init__()
+        self._controller = controller
+
+    def on_data_available(self, reader):
+        """Raise when there is new Status data available."""
+        info = fastdds.SampleInfo()
+        data = Status()
+        reader.take_next_sample(data, info)
+        self._controller.on_ddsrecorder_status.emit(
+            data.previous, data.current, data.info)
+
+
+class Controller(QObject):
+    """DDS Controller class."""
+
+    def __init__(
+        self,
+        dds_domain=0,
+        debug=False,
+        logger=None
+    ):
+        """Construct a DDS Controller."""
+        super().__init__()
+        # Initialize logger
+        self.logger = Logger(debug, logger)
+        self.logger.debug('Initializing DDS Recorder controller...')
+
+        # Check DDS Domain
+        if (not self.is_valid_dds_domain(dds_domain)):
+            raise ValueError(
+                'DDS Domain must be a number between 0 and 255.')
+
+        self.init_dds(dds_domain)
+
+    def __del__(self):
+        """Remove all dds entities in an orderly manner."""
+        self.delete()
+
+    on_ddsrecorder_discovered = pyqtSignal(str)
+    on_ddsrecorder_status = pyqtSignal(str, str, str)
+
+    def is_valid_dds_domain(self, dds_domain):
+        return ((dds_domain >= 0) and (dds_domain <= 255))
```

Comment:
```suggestion
        return ((dds_domain >= 0) and (dds_domain <= 232))
```
Better to have this value in a static variable rather than a magic number